### ISSUE TITLE
changes to lauch with current trial

### ISSUE THREAD

**Issue by azimut** [Shirakumo/ld45#1]
just trying trial and reporting what i found

minor
- missing cl-gamepad local-nickname
- use cl-gamepad:list-devices instead of cl-gamepad:devices

trial bug?
- hit an error on the line below, it was trying to run (attachment) on an input and throw an error. Filtering out the `input` types before the (find) resolve it for me
https://github.com/Shirakumo/trial/blob/72bc4f4d05fb098390278950d218155ff71f2bf2/pipeline.lisp#L160

**Comment by Shinmera** [comment]
I don't understand what you mean by the trial bug. The pipeline system works just fine.

**Comment by azimut** [comment]
Might be something ld45 does on trial internals?. I just ran `(ld45:launch)` and got the below error that I workaround by doing the remove/filter i mentioned above. 

```
There is no applicable method for the generic function
  #<STANDARD-GENERIC-FUNCTION TRIAL:ATTACHMENT (1)>
when called with arguments
  (#<INPUT #<GAUSSIAN-BLUR-PASS  {1005A67C73}>/PREVIOUS-PASS>).
   [Condition of type SB-PCL::NO-APPLICABLE-METHOD-ERROR]

Restarts:
 0: [RETRY] Retry calling the generic function.
 1: [ABORT] Don't set up the scene, leaving it empty.
 2: [RETRY] Retry SLY mREPL evaluation request.
 3: [*ABORT] Return to SLY's top level.
 4: [ABORT] abort thread (#<THREAD "sly-channel-1-mrepl-remote-1" RUNNING {1004CA5513}>)

Backtrace:
 0: ((:METHOD NO-APPLICABLE-METHOD (T)) #<STANDARD-GENERIC-FUNCTION TRIAL:ATTACHMENT (1)> #<INPUT #<GAUSSIAN-BLUR-PASS  {1005A67C73}>/PREVIOUS-PASS>) [fast-method]
 1: (SB-PCL::CALL-NO-APPLICABLE-METHOD #<STANDARD-GENERIC-FUNCTION TRIAL:ATTACHMENT (1)> (#<INPUT #<GAUSSIAN-BLUR-PASS  {1005A67C73}>/PREVIOUS-PASS>))
 2: (FIND :COLOR-ATTACHMENT0 (#<INPUT #1=#<GAUSSIAN-BLUR-PASS  {1005A67C73}>/PREVIOUS-PASS> #<OUTPUT #1#/COLOR>) :KEY #<STANDARD-GENERIC-FUNCTION TRIAL:ATTACHMENT (1)>)
 3: ((:METHOD PACK-PIPELINE (PIPELINE T)) #<WORLD 4 units {10056D32D3}> #<TRIAL-GLFW:CONTEXT {1005A11E43}>) [fast-method]
      Locals:
        TRIAL::PASS = #<GAUSSIAN-BLUR-PASS  {1005A67C73}>
        PASSES = (#<FOG-OF-WAR  {1005A64383}> #<GAUSSIAN-BLUR-PASS  {1005A67C73}> #<GAUSSIAN-BLUR-PASS  {1005A6CD13}> #<LIGHTING-PASS  {1005A6DEE3}>)
        PIPELINE = #<WORLD 4 units {10056D32D3}>
        TARGET = #<TRIAL-GLFW:CONTEXT {1005A11E43}>
        TRIAL::TEXSPECS = #((:WIDTH WIDTH :HEIGHT HEIGHT :TARGET :TEXTURE-2D ...) (:WIDTH WIDTH :HEIGHT HEIGHT :TARGET :TEXTURE-2D ...) (:WIDTH WIDTH :HEIGHT HEIGHT :TARGET :TEXTURE-2D ...))
        TEXTURES = #(#<TEXTURE 770x858 DEPTH-STENCIL {1005C748A3}> #<TEXTURE 770x858 RGBA {1006DA9543}> #<TEXTURE 770x858 RGBA {1006DAAD83}>)
 4: ((:METHOD BAKE (PIPELINED-SCENE)) #<WORLD 4 units {10056D32D3}>) [fast-method]
      Locals:
        SCENE = #<WORLD 4 units {10056D32D3}>
 5: ((:METHOD BAKE :AROUND (BAKABLE)) #<WORLD 4 units {10056D32D3}>) [fast-method]
      Locals:
        SB-PCL::.NEXT-METHOD-CALL. = #S(SB-PCL::FAST-METHOD-CALL :FUNCTION #<FUNCTION (SB-PCL::FAST-METHOD BAKE (PIPELINED-SCENE))> :PV NIL :NEXT-METHOD-CALL NIL :ARG-INFO (1))
        BAKABLE = #<WORLD 4 units {10056D32D3}>
 6: ((SB-PCL::EMF COMPUTE-RESOURCES) #<unused argument> #<unused argument> #<WORLD 4 units {10056D32D3}> #() #() #<HASH-TABLE :TEST EQ :COUNT 1 {1005A87CF3}>)
      Locals:
        SB-PCL::.ARG0. = #<WORLD 4 units {10056D32D3}>
        SB-PCL::.ARG1. = #()
        SB-PCL::.ARG2. = #()
        SB-PCL::.ARG3. = #<HASH-TABLE :TEST EQ :COUNT 1 {1005A87CF3}>
 7: (TRIAL::COMPUTE-RESOURCES-FOR #<WORLD 4 units {10056D32D3}>)
      Locals:
        THING = #<WORLD 4 units {10056D32D3}>
 8: ((:METHOD TRANSITION (NULL CONTAINER)) #<unused argument> #<WORLD 4 units {10056D32D3}>) [fast-method]
      Locals:
        TRIAL::TO = #<WORLD 4 units {10056D32D3}>
 9: ((:METHOD CHANGE-SCENE (TRIAL:MAIN SCENE)) #<MAIN {1004540953}> #<WORLD 4 units {10056D32D3}> :OLD NIL) [fast-method]
      Locals:
        #:.DEFAULTING-TEMP. = NIL
        TRIAL:MAIN = #<MAIN {1004540953}>
        #:N-SUPPLIED-6 = 1
        TRIAL::NEW = #<WORLD 4 units {10056D32D3}>
        TRIAL::OLD = NIL
10: ((:METHOD SETUP-RENDERING :AFTER (TRIAL:MAIN)) #<MAIN {1004540953}>) [fast-method]
      Locals:
        TRIAL:MAIN = #<MAIN {1004540953}>
11: ((SB-PCL::EMF SETUP-RENDERING) #<unused argument> #<unused argument> #<MAIN {1004540953}>)
      Locals:
        SB-PCL::.ARG0. = #<MAIN {1004540953}>
12: ((:METHOD INITIALIZE-INSTANCE :AFTER (DISPLAY)) #<MAIN {1004540953}> :WORLD NIL :LEVEL-NAMES (#1="credits/" "intro/" "map1/" "map2/" "map3/" #1#) :CLEAR-COLOR (VEC4 0.0 0.0 0.0 0.0) :TITLE "LD45" :WID..
      Locals:
        #:.DEFAULTING-TEMP. = NIL
        DISPLAY = #<MAIN {1004540953}>
        TRIAL::INITARGS = (:WORLD NIL :LEVEL-NAMES ("credits/" "intro/" "map1/" "map2/" "map3/" "credits/") :CLEAR-COLOR (VEC4 0.0 0.0 0.0 0.0) ...)
13: ((SB-PCL::EMF INITIALIZE-INSTANCE) #<unused argument> #<unused argument> #<MAIN {1004540953}> :WORLD NIL :LEVEL-NAMES (#1="credits/" "intro/" "map1/" "map2/" "map3/" #1#) :CLEAR-COLOR (VEC4 0.0 0.0 0...
14: ((:METHOD MAKE-INSTANCE (CLASS)) #<STANDARD-CLASS ORG.SHIRAKUMO.FRAF.LD45::MAIN> :WORLD NIL :LEVEL-NAMES (#1="credits/" "intro/" "map1/" "map2/" "map3/" #1#) :CLEAR-COLOR (VEC4 0.0 0.0 0.0 0.0) :TITLE..
15: (LAUNCH-WITH-CONTEXT MAIN :WORLD NIL :LEVEL-NAMES (#1="credits/" "intro/" "map1/" "map2/" "map3/" #1#))
      Locals:
        #:G11 = MAIN
        SB-DEBUG::MORE = (:WORLD NIL :LEVEL-NAMES ("credits/" "intro/" "map1/" "map2/" "map3/" "credits/"))
16: (TRIAL:LAUNCH MAIN :WORLD NIL :LEVEL-NAMES (#1="credits/" "intro/" "map1/" "map2/" "map3/" #1#))
      Locals:
        #:G2 = MAIN
        SB-DEBUG::MORE = (:WORLD NIL :LEVEL-NAMES ("credits/" "intro/" "map1/" "map2/" "map3/" "credits/"))
17: (SB-INT:SIMPLE-EVAL-IN-LEXENV (LAUNCH) #<NULL-LEXENV>)
18: (EVAL (LAUNCH))
19: ((LAMBDA NIL :IN SLYNK-MREPL::MREPL-EVAL-1))
 --more--
```

**Comment by Shinmera** [comment]
Hmm, you are correct. I'm just very confused how anything works at all at the moment, yet this triggers the issue.

**Comment by Shinmera** [comment]
Fixed in shirakumo/trial@46c67b2
